lib/remap-lang/src/parser.rs

```diff
@@ -146,9 +146,9 @@ impl<'a> Parser<'a> {
         }
 
         if let Some(expression) = expressions.last() {
-            let type_def = expression.type_def(&self.compiler_state);
+            let kind = expression.type_def(&self.compiler_state).scalar_kind();
 
-            if !type_def.kind.is_all() && type_def.kind.contains_regex() {
+            if !kind.is_all() && kind.contains_regex() {
```

Comment:
Technically, you can still _leak_ a regexp if its type is `any`.

We'll close this leak at some point as well, but for now, I considered it to be an edge-case (and one that has no implications, since Vector simply converts the regexp to its string representation), and the cost of plugging that hole was too great for now (denying returning `any` in a TRL script).